File: algorithm/number_theory/prime_number/sieve_of_eratosthenes/SieveOfEratosthenes.java
import java.util.*;

public class SieveOfEratosthenes {
  public static List<Integer> sieve(int n ){
    boolean[] isPrime = new boolean[n+1];
    Arrays.fill(isPrime, true);
    isPrime[0] = isPrime[1] = false;

    for(int i=2; i*i<=n ; i++){
      if(isPrime[i]){
        for(int j = i*i; j <=n; j+=i){
          isPrime[j] = false;
        }
      }
    }

    List<Integer> primes = new ArrayList<>();
    for(int i = 2;  i <= n ; i++){
      if(isPrime[i]){
        primes.add(i);
      }
    }
    return primes;
  }

  public static void main(String[] args){
    int n = 50;
    System.out.println(sieve(n));
  }
}

File: algorithm/number_theory/prime_number/sieve_of_eratosthenes/sieve_of_eratosthenes.py
#!/usr/bin/python3

def sieve_of_eratosthenes(n):
  #n까지의 모든 숫자를 소수라고 가정하고 True로 설정
  is_prime = [True] * (n+1)
  is_prime[0] = is_prime[1] = False # 0 과 1은 소수가 아님

  for i in range(2 , int(n**0.5) +1): #2부터 루트 N 까지 검사
    if is_prime[i]:   #i가 소수라면
      for j in range(i*i, n +1, i):   #i의 배수들을 False 처리
        is_prime[j] = False

  #소수 리스트 생성
  primes = [i for i in range(n+1) if is_prime[i]]
  return primes

n = 50
print(sieve_of_eratosthenes(n))